app/main.py

```diff
@@ -1 +1,110 @@
-# write your code here
+import os
+
+
+def move_file(command: str) -> None:
+    # Check if input command format is correct
+    if not is_input_data_correct(command):
+        return
+
+    # Get correct paths for source and destination files
+    origin_input_path, dest_input_path = command.split()[1:]
+    dest_short_path, dest_file_name = os.path.split(dest_input_path)
+    origin_short_path, origin_file_name = os.path.split(origin_input_path)
+    if not dest_file_name:
+        dest_full_path = os.path.join(dest_short_path, origin_file_name)
+    else:
+        dest_full_path = dest_input_path
+
+    # If new location is not provided:
+    # Rename source file
+    if not origin_short_path and not dest_short_path:
+        rename_file(origin_file_name, dest_file_name)
+        return
+
+    # If new location is provided:
+    # Open source file for reading
+    try:
+        origin_file = open(origin_input_path, "r")
+    except FileNotFoundError:
+        print(f"No source file with such name: {origin_input_path}")
+        return
+    except PermissionError:
+        print(f'"{origin_file_name}" is a folder')
+        return
+
+    # Create folder tree given in the destination path
+    try:
+        create_folder_tree(dest_short_path)
+    except OSError:
+        print(r'File name cannot contain the following characters: \/?*:"<>|')
+        return
+
+    # Create target file, copy data to it from source file, remove source file
+    dest_full_path = os.path.normpath(dest_full_path)
+    try:
+        with open(dest_full_path, "x") as dest_file:
```

Comment:
```suggestion
        with open(dest_full_path, "w") as dest_file:
```

Reply:
I use "x" to avoid modifying an existing file. Is it expected that if the user types the name of an existing file within the target path, it should be replaced with the source file? If so, it's not quite clear from the task description.

Reply:
You need to use open(dest_full_path, "w")